### PR TITLE
Correct plugin loader context for persistent

### PR DESCRIPTION
**Pull request by bcoca**
##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

plugin loader

### CODE REVIEW

File: lib/ansible/plugins/connection/persistent.py [hunk 1]
```diff
@@ -98,11 +99,20 @@ def _start_connection(self):
             raise AnsibleError("Unable to find location of 'ansible-connection'. "
                                "Please set or check the value of ANSIBLE_CONNECTION_PATH")
 
+        env = os.environ.copy()
+        env.update({
+                   'ANSIBLE_BECOME_PLUGINS': become_loader.print_paths(),
+                   'ANSIBLE_CLICONF_PLUGINS': cliconf_loader.print_paths(),
+                   'ANSIBLE_CONNECTION_PATH': connection_loader.print_paths(),
```
Comment (nitzmahone) [inline lib/ansible/plugins/connection/persistent.py]: Shouldn't this be `ANSIBLE_CONNECTION_PLUGINS`? `ANSIBLE_CONNECTION_PATH` is the path to `ansible-connection` itself, not the envvar used by the loader.

**Review by nitzmahone — CHANGES_REQUESTED** [review]
One (seemingly) incorrect envvar key- will also need to handle passing over the merged collections path at some point (but probably no need to hold for that)...

File: lib/ansible/plugins/connection/persistent.py (hunk 1 above)
Comment (bcoca) [reply lib/ansible/plugins/connection/persistent.py]: at least one of us had coffee today ...

**Comment by bcoca** [comment]
@nitzmahone eventually, for now this should be a fix for existing versions and keeping it separate allows for easy backport.

**Review by nitzmahone — APPROVED** [review]

**Review by Qalthos — APPROVED** [review]

**Comment by trishnaguha** [comment]
This introduced a bug in local connection plugin https://github.com/ansible/ansible/issues/55381.

**Comment by trishnaguha** [comment]
Update: There appears to be a bug in netconf plugin instead of here in the persistent plugin.

**Pull request merged**